2-mandatory/1-weather-report.js

```diff
@@ -12,7 +12,15 @@
 */
 
 function getTemperatureReport(cities) {
-    // TODO
+    output=[];
+    for (city of cities) {
+        temperature = temperatureService(city);
+        string = `The temperature in ${city} is ${temperature} degrees`;
+        output.push (string);
+
+    }
+    return output;
```

Comment:
Comment: Well done! 
Consider: You could also use iterate variable as below. 
function getTemperatureReport(cities) {
const temperatureReport = [];
    for(let i = 0; i < cities.length; i++) {
        const temperature = temperatureService(cities[i]);
        temperatureReport.push(`The temperature in ${cities[i]} is ${temperature} degrees`);
    }
    return temperatureReport;
}